codalab/worker_manager/slurm_batch_worker_manager.py

```diff
@@ -108,6 +109,14 @@ def __init__(self, args):
         # A set of newly submitted job id to keep tracking worker status, as worker might not be created right away.
         self.submitted_jobs = self.load_worker_jobs()
 
+        if args.password_file:
+            logger.info(
+                f"--password-file specified. Reading credentials from {args.password_file}..."
+            )
+            with open(args.password_file) as f:
+                os.environ["CODALAB_USERNAME"] = f.readline().strip()
+                os.environ["CODALAB_PASSWORD"] = f.readline().strip()
```

Comment:
Can you explain why we have to do this?

Reply:
We're using a temporary `CodaLabManager` for the worker and worker managers by default, so we don't write out the state to `state.json` when users log in. That's why running `nlp-cl-worker-manager` multiple times always prompts the user to log in. If we always read from the password file, we can both leave state.json untouched, and users can always start the worker manager without logging in.

Reply:
Could you please add a comment that explains some of this?